index.js

```diff
@@ -93,121 +114,153 @@ rl.on('line', function(line) {
                         reject();
                     });
                 }).catch(error => {
-                    console.log('Error creating group ' + groupData.name + ':', error);
+                    console.log('Error logging in admin user (' + admin.username + '):', error);
                     reject();
                 });
             }).catch(error => {
-                console.log('Error logging in admin user (' + admin.username + '):', error);
+                console.log('Error creating admin user (' + admin.username + '):', error);
                 reject();
             });
         }).catch(error => {
-            console.log('Error creating admin user (' + admin.username + '):', error);
+            console.log('Error logging in master user', error);
             reject();
         });
     });
 }
 
-function request(cmd, path, headers, postData) {
-    console.log('request, cmd:', cmd, ', path:', path);
-    var options = {
-        host: 'calendar.knilxof.org',
-        path: '/api/v2' + path,
-        port: '443',
-        headers: headers,
-        method: cmd
-    };
-
-    console.log('Creating \'request\' promise for', path, '...');
+function login(user) {
     return new Promise(function(resolve, reject) {
-        var req = https.request(options, (res) => {
-            if (res.statusCode !== 200 && res.statusCode !== 201 && res.statusCode !== 204) {
-                reject('request failed: ' + res);
+        var serverData;
+        loginRequest.post(
+            {
+              body: user
+            }, function (error, response, body) {
+                if (error || (response.statusCode !== 200 && response.statusCode !== 201 && response.statusCode !== 204)) {
+                    reject('login error:', response.statusCode + ': ', error);
+                }
             }
-            var result;
-            res.setEncoding('utf8');
-            res.on('data', (data) => {
-                result = JSON.parse(data); });
-            res.on('end', () => {
-                resolve(result);
-            });
+        ).on('data', function(data) {
+            serverData = JSON.parse(data);
+        }).on('end', function() {
+            resolve(serverData);
+        });
+    });
+}
+
+function createUser(user, token) {
+    return new Promise(function(resolve, reject) {
+        var serverData;
+        createUserRequest.post(
+            {
+              body: user,
+              headers: {'Authorization': 'Bearer ' + token}
+            }, function (error, response, body) {
+                if (error || (response.statusCode !== 200 && response.statusCode !== 201 && response.statusCode !== 204)) {
+                    reject('createUser error:', response.statusCode + ': ', error);
+                }
+            }
+        ).on('data', function(data) {
+            serverData = JSON.parse(data);
+        }).on('end', function() {
+            resolve(serverData);
+        });
```

Comment:
maybe these 2 callbacks could go to the previous callback too, as it's really simple here (we don't need to stream the response).